auth/api/iam/api_mitzxnuts.go

```diff
@@ -0,0 +1,464 @@
+package iam
+
+import (
+	"context"
+	crypt "crypto"
+	"encoding/json"
+	"errors"
+	"github.com/google/uuid"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/auth/log"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	http2 "github.com/nuts-foundation/nuts-node/http"
+	"github.com/nuts-foundation/nuts-node/storage"
+	"github.com/nuts-foundation/nuts-node/vcr/holder"
+	"github.com/nuts-foundation/nuts-node/vcr/pe"
+	"github.com/nuts-foundation/nuts-node/vcr/verifier"
+	"github.com/nuts-foundation/nuts-node/vdr/didweb"
+	"github.com/nuts-foundation/nuts-node/vdr/resolver"
+	"io"
+	"maps"
+	"net/http"
+	"net/url"
+	"slices"
+	"time"
+)
+
+const oid4vicSessionValidity = 15 * time.Minute
+
+type Oid4vciSession struct {
+	HolderDid   string
+	IssuerDid   string
+	RedirectUrl string
+	RedirectUri string
+}
+
+func (r Wrapper) StartOid4vciIssuance(ctx context.Context, request StartOid4vciIssuanceRequestObject) (StartOid4vciIssuanceResponseObject, error) {
```

Comment:
for OpenID4VP I put the creation of the redirectURL in the client. The client has access to crypto and metadata methods. Architectural speaking it would be in the 'service' layer, but with only server and client layers available it's better suited in the 'client' layer.

Reply:
It is moved to /internal and returns a JSON object with a redirect_uri in it.